### PR TITLE
Add e2e tests for service with concurrency options

### LINKED ISSUE #253: Add integration tests for concurrency options of service create and update
For `service create` and `service update`, we are missing coverage for following options
```
      --concurrency-limit int    Hard Limit of concurrent requests to be processed by a single replica.
      --concurrency-target int   Recommendation for when to scale up based on the concurrent number of incoming request. Defaults to --concurrency-limit when given.
```

### PR DESCRIPTION
**Pull request by toVersus**
Check the behaviour of creating/updating service with concurrency options

Fixes #253

### CODE REVIEW

File: test/e2e/basic_workflow_test.go [hunk 1]
```diff
@@ -87,6 +93,19 @@ func testServiceCreate(t *testing.T, k kn, serviceName string) {
 	}
 }
 
+func testServiceCreateWithOptions(t *testing.T, k kn, serviceName string, args []string) {
```
Comment (navidshaikh) [inline test/e2e/basic_workflow_test.go]: probably rename args with options ?

File: test/e2e/basic_workflow_test.go [hunk 2]
```diff
@@ -87,6 +93,19 @@ func testServiceCreate(t *testing.T, k kn, serviceName string) {
...
+	command := []string{"service", "create", fmt.Sprintf("%s", serviceName), "--image", KnDefaultTestImage}
```
Comment (navidshaikh) [inline test/e2e/basic_workflow_test.go]: just use the serviceName directly here?

File: test/e2e/basic_workflow_test.go [hunk 3]
```diff
@@ -87,6 +93,19 @@ func testServiceCreate(t *testing.T, k kn, serviceName string) {
...
+	command = append(command, args...)
+	out, err := k.RunWithOpts(command, runOpts{NoNamespace: false})
+	if err != nil {
+		t.Fatalf(fmt.Sprintf("Error executing 'kn service create' command. Error: %s", err.Error()))
+	}
+
+	if !strings.Contains(out, fmt.Sprintf("Service '%s' successfully created in namespace '%s'.", serviceName, k.namespace)) {
```
Comment (navidshaikh) [inline test/e2e/basic_workflow_test.go]: In other tests, we are testing the output string to contain specific words, for eg:
https://github.com/knative/client/blob/9fb6a436e81d69935a67cc5b245cc634b2ede55b/test/e2e/revision_workflow_test.go#L48

Comment (toVersus) [reply test/e2e/basic_workflow_test.go]: Thanks! I changed to use assert here, but it seems that raw `strings.Contains` is used for asserting output text in other e2e tests. Can I follow up to fix them in separate PR?

Comment (navidshaikh) [reply test/e2e/basic_workflow_test.go]: sure!

File: test/e2e/service_advance_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		t.Fatalf(fmt.Sprintf("Error executing 'kn service describe' command. Error: %s", err.Error()))
+	}
+
+	expectedOutput := fmt.Sprintf("containerConcurrency: %s", concurrencyLimit)
+	assert.Check(t, util.ContainsAll(out, expectedOutput))
+}
+
+func testServiceDescribeConcurrencyTarget(t *testing.T, k kn, serviceName, concurrencyTarget string) {
+	out, err := k.RunWithOpts([]string{"service", "describe", serviceName}, runOpts{NoNamespace: false})
+	if err != nil {
```
Comment (rhuss) [inline test/e2e/service_advance_test.go]: You can use `assert.NilError()` to avoid that explicit check.

File: test/e2e/service_advance_test.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		t.Fatalf(fmt.Sprintf("Error executing 'kn service create' command. Error: %s", err.Error()))
+	}
+
+	assert.Check(t, util.ContainsAll(out, "Service", serviceName, "successfully created in namespace", k.namespace, "OK"))
+}
+
+func testServiceDescribeConcurrencyLimit(t *testing.T, k kn, serviceName, concurrencyLimit string) {
+	out, err := k.RunWithOpts([]string{"service", "describe", serviceName}, runOpts{NoNamespace: false})
+	if err != nil {
```
Comment (rhuss) [inline test/e2e/service_advance_test.go]: `assert.NilError()`

File: test/e2e/service_advance_test.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/client/pkg/util"
+	"gotest.tools/assert"
+)
+
+func TestServiceAdvance(t *testing.T) {
```
Comment (rhuss) [inline test/e2e/service_advance_test.go]: Please call it more specific like `TestServiceOptions` and also rename the file to `service_options_test.go`

I would then add also here the test for minScale and maxScale.

**Comment by toVersus** [comment]
@rhuss Thanks for your review! I addressed the comments.

**Pull request merged**